src/process_utility.c

```diff
@@ -2581,19 +2582,50 @@ validate_set_not_null(Hypertable *ht, Oid chunk_relid, void *arg)
 	Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, true);
 	if (ts_chunk_is_compressed(chunk) && !ts_is_hypercore_am(chunk->amoid))
 	{
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("operation not supported on compressed chunks not using the "
-						"\"hypercore\" table access method"),
-				 errdetail("Chunk %s.%s is using the heap table access method and has compressed "
-						   "data.",
-						   NameStr(chunk->fd.schema_name),
-						   NameStr(chunk->fd.table_name)),
-				 errhint("Either decompress all chunks of the hypertable or use \"ALTER TABLE "
-						 "%s.%s SET ACCESS METHOD hypercore\" on all chunks to change access "
-						 "method.",
-						 NameStr(chunk->fd.schema_name),
-						 NameStr(chunk->fd.table_name))));
+		StringInfoData command;
+		AlterTableCmd *cmd = (AlterTableCmd *) arg;
+		Chunk *comp_chunk = ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, true);
+
+		initStringInfo(&command);
+		appendStringInfo(&command,
+						 "SELECT EXISTS(SELECT FROM %s.%s WHERE %s IS NULL",
```

Comment:
When the compressed column has a null value, it means a default, so probably shouldn't be an OR.

Reply:
we also just found out it's not true, and we can't distinguish missing from default on the sql level..